ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/categories/AppCategoriesPresenterImpl.java

```diff
@@ -362,33 +296,36 @@ public void onSuccess(String result) {
     }
 
     @Override
-    public void onRemove(StoreRemoveEvent<App> event) {
-        // When the list store removes something
-        AppCategory appCategory = getSelectedAppCategory();
-        if (appCategory == null) {
-            return;
-        }
-        updateAppCategoryAppCount(appCategory, event.getSource().getAll().size());
+    public HandlerRegistration addAppCategorySelectedEventHandler(AppCategorySelectionChangedEvent.AppCategorySelectionChangedEventHandler handler) {
+        return ensureHandlers().addHandler(AppCategorySelectionChangedEvent.TYPE, handler);
     }
 
-    void addAppCategories(AppCategory parent, List<AppCategory> children) {
+    @Override
+    public void onAppCategorySelectionChanged(AppCategorySelectionChangedEvent event) {
+        fireEvent(event);
+    }
+
+    void addAppCategories(TreeStore<AppCategory> treeStore,
+                          AppCategory parent,
+                          List<AppCategory> children) {
         if ((children == null)
                 || children.isEmpty()) {
             return;
         }
         if (parent == null) {
+            treeStore.clear();
             treeStore.add(children);
         } else {
             treeStore.add(parent, children);
         }
 
         for (AppCategory ag : children) {
-            addAppCategories(ag, ag.getCategories());
+            addAppCategories(treeStore, ag, ag.getCategories());
         }
     }
 
     AppCategory findAppCategoryByName(String name) {
-        for (AppCategory appCategory : treeStore.getAll()) {
+        for (AppCategory appCategory : workspaceView.getTree().getStore().getAll()) {
```

Comment:
Since I seem to keep needing the view's treeStore I should probably have a copy in the presenter?